src/hubinit/repulishjob.bal

```diff
@@ -0,0 +1,35 @@
+import ballerina/config;
+import ballerina/log;
+import ballerina/runtime;
+import ballerina/websub;
+import mosip/repository;
+import mosip/services;
+
+public type RepublishJob object {
+
+    public function __init() {
+    }
+  
+    service republishservice = service {
+
+        resource function onTrigger(services:HubServiceImpl hubServiceImpl, websub:Hub webSubHub) {
+            repository:RestartRepublishContentModel[] unsentMessages = hubServiceImpl.getUnsentMessages(config:getAsString("mosip.hub.restart_republish_time_offset"));
+            if (unsentMessages.length() > 0) {
+                foreach var unsentMessage in unsentMessages {
+                    var publishResponse = webSubHub.publishUpdate(unsentMessage.topic, unsentMessage.message.toBytes());
```

Comment:
Will this not cause the duplicate publish when running this job continiously ? since there wil be some time gap between a message reaching the message store and the same reaching the success or failed delivery table. If the job run inbetween this time, it will republish the same message even though it is still available in the queue

Reply:
yes we will be giving time large enough like a day or two so that only after that it does the job.

Reply:
ok resolved